### PR TITLE
DLPX-88137 Upgrade sshj with Raul's mod dlpx73623

### PR DESCRIPTION
**Pull request by lschetanrao**
JIRA : https://delphix.atlassian.net/browse/DLPX-88137

# Problem
Updating sshj with custom changes.

# Solution
This PR contains following changes 

1. Clean cherry-pick of https://github.com/delphix/sshj/pull/6 on top of the latest release, 0.36.0.
2. Fixed unit test cases by replacing junit with jupiter junit because SSHJ has removed junit dependency. Please check this commit https://github.com/delphix/sshj/pull/7/commits/af057500d31b701a741422919dc3e0e653e9add4

Published the JAR on artifactory as version 0.36.0-dlpx73623. https://artifactory.delphix.com/ui/repos/tree/General/appstack-internal-services/com/hierynomus/sshj/0.36.0-dlpx73623/sshj-0.36.0-dlpx73623.jar

Published the POM file by copying the pom from https://repo1.maven.org/maven2/com/hierynomus/sshj/0.36.0/sshj-0.36.0.pom and replacing the version from **0.36.0** to **0.36.0-dlpx73623**

### CODE REVIEW

File: src/test/java/net/schmizz/sshj/common/CircularBufferTest.java [hunk 1]
```diff
@@ -159,42 +160,53 @@ public void shouldAllowWritingMaxRemainingCapacityAfterWrappingAround() throws C
...
 
-        final int initiallyWritten = 10;
-        buffer.putRawBytes(new byte[initiallyWritten], 0, initiallyWritten);
+            final int initiallyWritten = 10;
+            buffer.putRawBytes(new byte[initiallyWritten], 0, initiallyWritten);
 
-        // Max remaining capacity is always one less than the remaining buffer size.
-        int maxRemainingCapacity = buffer.maxPossibleRemainingCapacity();
-        assertEquals(buffer.length() - 1 - initiallyWritten, maxRemainingCapacity);
+            // Max remaining capacity is always one less than the remaining buffer size.
```
Comment (VenkatanadhanG) [inline src/test/java/net/schmizz/sshj/common/CircularBufferTest.java:171]: [nit]: Check for indentation issues

**Comment by sebroy** [comment]
@rasantel @lschetanrao Is there any hope of upstreaming our fix so that we don't have to maintain this fork?

File: src/test/java/net/schmizz/sshj/common/CircularBufferTest.java (hunk 1 above)
Comment (lschetanrao) [reply src/test/java/net/schmizz/sshj/common/CircularBufferTest.java:171]: I think indentation is correct.

**Review by VenkatanadhanG — COMMENTED** [review]
Waiting for ab pre push  and dx test with new version of sshj and manual testing to confirmation on app-gate.

**Review by rasantel — APPROVED** [review]

**Comment by rasantel** [comment]
> @rasantel @lschetanrao Is there any hope of upstreaming our fix so that we don't have to maintain this fork?

@sebroy Our fix converts the buffer class to a circular buffer only for the parts that we use, so there are now two classes - the old one used in other parts of sshj and the new one. I was hoping to find time to complete the conversion and get rid of the old buffer  before submitting, but in practice there's always something else taking my time. Finishing the conversion is not necessarily trivial because of the impact throughout the entire library. I suppose I could submit the partial conversion and see what happens.

**Comment by rasantel** [comment]
@sebroy By the way, I already upstreamed another race condition fix, https://github.com/hierynomus/sshj/pull/851, which allowed us to be able to upgrade sshj again.

**Comment by sebroy** [comment]
> @sebroy By the way, I already upstreamed another race condition fix, [hierynomus#851](https://github.com/hierynomus/sshj/pull/851), which allowed us to be able to upgrade sshj again.

@rasantel that's excellnt, thank you.

**Review by VenkatanadhanG — APPROVED** [review]

**Pull request merged**